pkg/controllers/taint/clustertaintpolicy_controller.go

```diff
@@ -57,28 +58,30 @@ type ClusterTaintPolicyController struct {
 // The Controller will requeue the Request to be processed again if an error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (c *ClusterTaintPolicyController) Reconcile(ctx context.Context, req controllerruntime.Request) (controllerruntime.Result, error) {
-	klog.V(4).Infof("Reconciling Cluster %s.", req.Name)
-
-	clusterObj := &clusterv1alpha1.Cluster{}
-	if err := c.Client.Get(ctx, req.NamespacedName, clusterObj); err != nil {
-		klog.Errorf("Failed to get Cluster(%s), error: %v", req.Name, err)
+	klog.V(4).Infof("Reconciling ClusterTaintPolicy %s.", req.Name)
+	policy := &policyv1alpha1.ClusterTaintPolicy{}
+	if err := c.Client.Get(ctx, req.NamespacedName, policy); err != nil {
+		klog.Errorf("Failed to get ClusterTaintPolicy(%s), error: %v", req.Name, err)
 		return controllerruntime.Result{}, client.IgnoreNotFound(err)
 	}
 
-	if !clusterObj.DeletionTimestamp.IsZero() {
-		klog.V(4).Infof("Cluster(%s) is deleting.", req.Name)
+	if !policy.DeletionTimestamp.IsZero() {
+		klog.V(4).Infof("ClusterTaintPolicy(%s) is deleting.", req.Name)
 		return controllerruntime.Result{}, nil
 	}
 
-	clusterTaintPolicyList := &policyv1alpha1.ClusterTaintPolicyList{}
+	clusterList := &clusterv1alpha1.ClusterList{}
 	listOption := &client.ListOptions{UnsafeDisableDeepCopy: ptr.To(true)}
-	if err := c.Client.List(ctx, clusterTaintPolicyList, listOption); err != nil {
-		klog.Errorf("Failed to list ClusterTaintPolicy, error: %v", err)
+	if err := c.Client.List(ctx, clusterList, listOption); err != nil {
+		klog.Errorf("Failed to list Cluster, error: %v", err)
 		return controllerruntime.Result{}, err
 	}
 
-	clusterCopyObj := clusterObj.DeepCopy()
-	for _, policy := range clusterTaintPolicyList.Items {
+	var errs []error
+	for index := range clusterList.Items {
+		clusterObj := &clusterList.Items[index]
+		clusterCopyObj := clusterObj.DeepCopy()
```

Comment:
Maybe we can skip the deep copy here and at the same time not use the `UnsafeDisableDeepCopy` option when listing the Clusters above.